app/src/main/java/com/infomaniak/mail/ui/newMessage/NewMessageViewModel.kt

```diff
@@ -730,31 +757,52 @@ class NewMessageViewModel @Inject constructor(
         }.onFailure(Sentry::captureException)
     }
 
-    fun executeDraftActionWhenStopping(
-        action: DraftAction,
-        isFinishing: Boolean,
-        isTaskRoot: Boolean,
-        subjectValue: String,
-        uiBodyValue: String,
-        startWorkerCallback: () -> Unit,
-    ) = globalCoroutineScope.launch(ioDispatcher) {
+    fun saveBodyAndSubject(subject: String, html: String) {
+        globalCoroutineScope.launch {
```

Comment:
Any reason to use `globalCoroutineScope` instead of the `viewModelScope` ?

Reply:
Yeah, the code is executed (amongst other situations) when the activity stops which in turns stop the view model and its scope but the `exportHtml()`, being async, can call the callback after this point. In this situation, if we're using the `viewModelScope`, the code inside the scope will never be called because the scope has already been closed